Remove useless Y param for POST and PUT

LukeMarlin opened the pull request:
Fixes #33 

While this seemed like a better idea to have type checking instead of
plain `object`, in reality it is just an added pain that gives no
benefit as the only thing done here is serializing with json.NET which
works with `object` itself anyway.

Signed-off-by: Luke Marlin <luke.marlin@viacesi.fr>

Commit:
Remove useless Y param for POST and PUT

While this seemed like a better idea to have type checking instead of
plain `object`, in reality it is just an added pain that gives no
benefit as the only thing done here is serializing with json.NET which
works with `object` itself anyway.

Signed-off-by: Luke Marlin <luke.marlin@viacesi.fr>

Commit:
Restore timeout test value

Signed-off-by: Luke Marlin <luke.marlin@viacesi.fr>

LukeMarlin commented:
@Namaneo, care to join? :)

Namaneo commented:
With those changes, I'm not sure it's a good idea to keep two generic methods with only one different parameter. 

Let's say I call `c.PutAsync<Contact>("/me/contact", "Fake content")` : currently as a user, I'm not quite sure if `"Fake content"` will be serialized or not, and even if I know what's going on under the hood, their will be cases where I would like to serialize this string. I don't think this is currently possible (other than manually serializing the string beforehand)?

Only keeping one non-generic method for when the user wants to manipulate raw data, and one generic method for when he wants built-in serialization might be a better option. What do you think?

Commit:
Make more explicit methods to avoid confusion

  - Passing a raw string that is not meant to be serialized is not done
  in [Post|Put]StringAsync methods.
  - Remove now useless CallAsync override
  - Add missing timeout passing
  - Change test methods names to be more consistent
  - Add more tests for previously uncovered overrides

Signed-off-by: Luke Marlin <luke.marlin@viacesi.fr>

LukeMarlin commented:
@Namaneo, comment was spot on.
I made some changes to make the behaviour more obvious: raw-strings methods have a specific name, and the param name also reflect that they should already be json.
This also helped on removing some methods and therefore simplifying a bit the code.

Does that look any better?

P.S: quite tired, so I might have made some mistakes. I'll take another look tomorrow anyway.

Commit:
Fix timeout testing

Signed-off-by: Luke Marlin <luke.marlin@viacesi.fr>
